underlay/build.gradle

```diff
@@ -1,50 +1,78 @@
 plugins {
     id 'java'
     id 'java-test-fixtures'
+    id 'pmd'
+
     id 'tanagra.java-conventions'
+    id 'org.springframework.boot' version '3.2.5'
+    id 'io.spring.dependency-management' version '1.1.5'
```

Comment:
Didn't notice this the first time around, but why do we need a Spring dependency in the `underlay` sub-project? I think we should only need this in the `service` and perhaps `client` sub-projects. If this is required because of the dependency on `terra-common-lib`, and we can't make it work without it, then maybe we should look into removing that dependency. I think (not certain, but will check) that we're only using that library for some common exceptions.